src/components/resource/_fields/custom-embargo/custom-embargo-fields.js

```diff
@@ -32,124 +35,119 @@ function validateEmbargoValue(value) {
   return error;
 }
 
-function validateEmbargoUnit(value, { customEmbargoValue }) {
+function validateEmbargoUnit(embargoUnit, { customEmbargoPeriod }) {
   let error;
+  const embargoValueIsSet = customEmbargoPeriod.length && customEmbargoPeriod[0].embargoValue > 0;
 
-  if (customEmbargoValue > 0 && !value) {
+  if (embargoValueIsSet && !embargoUnit) {
     error = <FormattedMessage id="ui-eholdings.validate.errors.embargoPeriod.unit" />;
   }
 
   return error;
 }
 
 export default class CustomEmbargoFields extends Component {
-  static propTypes = {
-    change: PropTypes.func.isRequired,
-    initial: PropTypes.object,
-    showInputs: PropTypes.bool
-  };
-
-  static defaultProps = {
-    initial: []
-  };
-
-  state = {
-    showInputs: this.props.showInputs
-  };
-
-  clearValues = () => {
-    this.props.change('customEmbargoValue', 0);
-    this.props.change('customEmbargoUnit', '');
-    this.toggleInputs();
+  renderEmbargoUnitField(fields) {
+    return (
+      <div
+        data-test-eholdings-custom-embargo-select
+        className={styles['custom-embargo-select']}
+      >
+        <FormattedMessage id="ui-eholdings.label.selectTimePeriod">
+          {placeholder => (
+            <Field
+              name={`${fields.name}[0].embargoUnit`}
```

Comment:
If I extract it in the arguments, it can be unclear from `renderEmbargoUnitField({name}) {` what name is. If I do `renderEmbargoUnitField({name: fieldsName}) {` or `renderEmbargoUnitField(fields: { name }) {`, it won't make the code shorted. For me all of the above options, including the one currently implemented,  the are OK. I don't know why to focus on this.